plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/expression/ExpEmbedding.kt

```diff
@@ -374,9 +374,9 @@ data class FieldAccess(val receiver: ExpEmbedding, val field: FieldEmbedding) :
     }
 
     private fun unfoldHierarchy(receiverWrapper: ExpEmbedding, ctx: LinearizationContext) {
-        val hierarchyPath = (receiver.type as? ClassTypeEmbedding)?.hierarchyUnfoldPath(field.name)
+        val hierarchyPath = (receiver.type as? ClassTypeEmbedding)?.details?.hierarchyUnfoldPath(field.name)
         hierarchyPath?.forEach { classType ->
-            val predAcc = classType.sharedPredicateAccessInvariant().fillHole(receiverWrapper)
+            val predAcc = classType.details.sharedPredicateAccessInvariant().fillHole(receiverWrapper)
```

Comment:
Hmm, I think `.details` can be omitted here since you've overridden this method anyway.
In general, should we hide those `.details` calls under some extension functions of `TypeEmbedding`?

Reply:
I'm not sure `TypeEmbedding` is the best place; I'd prefer us to fail explicitly when we need a `ClassTypeEmbedding` but have something else.  We could make functions that forward things to details, but I'm not sure there's much benefit; `details` is not meant to be private, and being aware you're using the details can be seen as a plus.